app/controllers/application_controller.rb

```diff
@@ -317,15 +317,23 @@ def reauthn?
 
   def confirm_two_factor_authenticated(id = nil)
     return prompt_to_sign_in_with_request_id(id) if user_needs_new_session_with_request_id?(id)
+
     authenticate_user!(force: true)
-    return prompt_to_setup_mfa unless two_factor_enabled?
-    return prompt_to_verify_mfa unless user_fully_authenticated?
-    return prompt_to_setup_mfa if service_provider_mfa_policy.
-                                  user_needs_sp_auth_method_setup?
-    return prompt_to_setup_non_restricted_mfa if two_factor_kantara_enabled?
-    return prompt_to_verify_sp_required_mfa if service_provider_mfa_policy.
-                                               user_needs_sp_auth_method_verification?
+
+    if !two_factor_enabled?
+      return prompt_to_setup_mfa
+    elsif !user_fully_authenticated?
+      return prompt_to_verify_mfa
+    elsif service_provider_mfa_policy.user_needs_sp_auth_method_setup?
+      return prompt_to_setup_mfa
+    elsif two_factor_kantara_enabled?
+      return prompt_to_setup_non_restricted_mfa
+    elsif service_provider_mfa_policy.user_needs_sp_auth_method_verification?
+      return prompt_to_verify_sp_required_mfa
+    end
```

Comment:
@zachmargolis I'm on the fence here. I think this is better, but it doesn't _look_ as good.

Also someone should double-check my logic since I just some `unless`es into `if !`s and accidentally inverting logic here would probably be quite bad.

Reply:
- I spot checked the `unless` -> `if !` and that looks good
- I think this `if`/`else` looks great